### PR TITLE
feat: Add transpose support for small homogeneously typed DataFrames.

### PR DESCRIPTION
**Pull request by TrevorBergeron**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: third_party/bigframes_vendored/pandas/core/frame.py [hunk 1]
```diff
@@ -93,6 +93,84 @@ def values(self) -> np.ndarray:
...
+        """
+        raise NotImplementedError(constants.ABSTRACT_METHOD_ERROR_MESSAGE)
+
+    def transpose(self) -> DataFrame:
+        """
+        Transpose index and columns.
+
+        Reflect the DataFrame over its main diagonal by writing rows as columns
+        and vice-versa. The property :attr:`.T` is an accessor to the method
+        :meth:`transpose`.
```
Comment (milkshakeiii) [inline third_party/bigframes_vendored/pandas/core/frame.py:131]: Can we add something here about the dtypes requirements? (Columns must be coerceable to a common type) (I know it's below but seems important enough to be up top.)

File: tests/system/small/test_dataframe.py [hunk 2]
```diff
@@ -2465,6 +2465,24 @@ def test_df_describe(scalars_dfs):
     ).all()
 
 
+def test_df_transpose():
```
Comment (milkshakeiii) [inline tests/system/small/test_dataframe.py:2468]: If it's a cheap test, we could add a check that we raise TypeError for mixing strs and floats.

File: bigframes/core/blocks.py [hunk 3]
```diff
@@ -1570,6 +1575,52 @@ def melt(
             index_columns=[index_id],
         )
 
+    def transpose(self) -> Block:
+        """Transpose the block. Will fail if dtypes incompatible or too many rows"""
```
Comment (milkshakeiii) [inline bigframes/core/blocks.py:1579]: Suggest we change "dtypes incompatible" to "dtypes cannot be coerced all to a common type."

**Review by milkshakeiii — APPROVED** [review]
LGTM, just suggested we make it more clear in the docstrings exactly what the dtype restriction is.

File: third_party/bigframes_vendored/pandas/core/frame.py (hunk 1 above)
Comment (TrevorBergeron) [reply third_party/bigframes_vendored/pandas/core/frame.py:133]: added

File: bigframes/core/blocks.py (hunk 3 above)
Comment (TrevorBergeron) [reply bigframes/core/blocks.py]: done

File: tests/system/small/test_dataframe.py (hunk 2 above)
Comment (TrevorBergeron) [reply tests/system/small/test_dataframe.py:2468]: added

**Pull request merged**